scripts/build/setup/termux_setup_ninja.sh

```diff
@@ -14,7 +14,10 @@ termux_setup_ninja() {
 		fi
 		export PATH=$NINJA_FOLDER:$PATH
 	else
-		if [ "$(dpkg-query -W -f '${db:Status-Status}\n' ninja 2>/dev/null)" != "installed" ]; then
+		local NINJA_PKG_VERSION=$(bash -c ". $TERMUX_SCRIPTDIR/packages/ninja/build.sh; echo \$TERMUX_PKG_VERSION")
+		if ([ ! -e "$TERMUX_BUILT_PACKAGES_DIRECTORY/ninja" ] ||
+		    [ "$(cat "$TERMUX_BUILT_PACKAGES_DIRECTORY/ninja")" != "$NINJA_PKG_VERSION" ]) &&
+		   [ "$(dpkg-query -W -f '${db:Status-Status}\n' ninja 2>/dev/null)" != "installed" ]; then
```

Comment:
Once I build ninja above, only checking dpkg-query will still stop any other build scripts from using it. Maybe we should do away with .built-packages altogether on-device and simply install the newly built packages from debs/ with dpkg instead, then only check using dpkg-query on device.